static/panes/editor.ts

```diff
@@ -1132,6 +1127,16 @@ export class Editor extends MonacoPane<monaco.editor.IStandaloneCodeEditor, Edit
         // @ts-expect-error: See above comment maybe
         this.editor.getModel()?.pushEditOperations(viewState?.cursorState ?? null, [operation], nullFn);
         this.numberUsedLines();
+    }
+
+    updateSource(newSource: string): void {
+        // Create something that looks like an edit operation for the whole text
+        const operation = {
+            range: this.editor.getModel()!.getFullModelRange(),
```

Comment:
is the change from `?` to `!` intentional here? My instinct is it's OK but I'm still learning much of the typescript stuff here :)